hash/hash_comparator.go

```diff
@@ -69,6 +89,26 @@ func CompareBcrypt(_ context.Context, password []byte, hash []byte) error {
 	return nil
 }
 
+func CompareSHA256Crypt(_ context.Context, password []byte, hash []byte) error {
+	decoder := crypt.NewDecoder()
```

Comment:
Would it make sense to initialize the decoder once with the supported algorithms (md5, sha256, sha512) and then just use it here?